interview-questions.md

```diff
@@ -7,46 +7,94 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. In a model called Animal that has_many Sightings, what is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: The name of the foreign key is animal_id and it would be a part of the sightings model.
 
+  Researched answer: Part one of my answer holds correct due to the naming convention of foreign keys. Foreign keys are named after the modelname with the has_many association (modelname_id). Foreign keys also reside in the model with the belongs_to association. 
 
 
 2. Which routes must always be passed params and why?
 
-  Your answer:
+  Your answer: create, update, delete methods must pass params
+
+  Researched answer: According to the lesson plan on rails forms, the purpose of creating forms is to permit users to "add, update, and delete information" without having to use the terminal. In order to accomplish this, we need to add a route that will show our user a page where they can enter information in a form and a route that will post the information to the database. This is done by passing params via the get/post methods. 
 
-  Researched answer:
+  Class Answer: show, update, destroy methods in controller
 
 
 
 3. Based on your knowledge of Rails conventions, write a route for a webpage called "game" that takes in a parameter called "guess" with a controller called "main".
 
-  Your answer:
+Notes: "A routes.rb file in the config folder is created when you run rails new. This file is intended to house all valid urls in your application"
+Rails naming convention includes all of the following:
 
+a.  http verb we are sending to the server via the url (get, post, put, patch, delete)
+b.  '/url'
+      =>
+c   'name of controller#'
+d.  '#name of method found in controller' 
+  Your answer: 
+  get '/game/:guess' => 'main#game'
+  
+  Class Answer: get 'main/game/:guess' => 'main#game'
 
 
-4. Name three rails generator commands. What is created by each?
 
-  Your answer:
-
-  Researched answer:
+4. Name three rails generator commands. What is created by each?
 
+  Your answer: I can recall two commands (1&4) but will need to research my notes to provide a third command:
+  1. $ rails generate model ModelName
+  2. $ rails generate migration ModificationOnModel
+  3. $ rails generate controller main
+  4. $ rails generate resource
```

Comment:
Excellent.  Again, I think I would also consider adding more information, to the best of your ability, on what is created by each as well.